prime-angular-frontend/src/app/modules/enrolment/shared/services/enrolment-form-state.service.ts

```diff
@@ -168,14 +170,36 @@ export class EnrolmentFormStateService extends AbstractFormStateService<Enrolmen
 
   /**
    * @description
-   * Check for the requirement of at least one certification, or one job.
+   * Check for the requirement of at least one certification, or one obo site/job.
    */
   public hasCertificateOrJob(): boolean {
-    const jobs = this.jobsForm.get('jobs') as FormArray;
+    const oboSites = this.jobsForm.get('oboSites') as FormArray;
     const certifications = this.regulatoryFormState.certifications;
+    const enrolleeHealthAuthorities = this.careSettingsForm.get('enrolleeHealthAuthorities') as FormArray;
+    let hasOboSiteForEveryHA = true;
+    // Using `for` loop rather than `every()` method for ease of debugging
+    for (let i = 0; i < enrolleeHealthAuthorities.controls.length; i++) {
+      const checkbox = enrolleeHealthAuthorities.controls[i];
+      // For every selected Health Authority ...
+      if (checkbox.value === true) {
+        let foundMatchingHAOboSite = false;
+        // ... there must be at least one Obo Site for that Health Authority
+        for (let j = 0; j < oboSites.controls.length; j++) {
```

Comment:
If you're checking for at least one you don't need a `for` loop since JS is functional it has a bunch of methods for doing checks for a single and exit - `oboSites.controls.some(predicateFunction)`.  It will exit with true if your predicate passes at any point and immediately exit, otherwise false.  I've added this to the example above.

Reply:
I'll make these updates quickly in a separate PR